lms/djangoapps/courseware/courses.py

```diff
@@ -519,6 +519,34 @@ def get_course_assignment_date_blocks(course, user, request, num_return=None,
     return date_blocks
 
 
+@request_cached()
+def get_course_blocks_completion_summary(course_key, user):
+    if not user.id:
+        return []
+    store = modulestore()
+    course_usage_key = store.make_course_usage_key(course_key)
+    block_data = get_course_blocks(user, course_usage_key, allow_start_dates_in_future=True, include_completion=True)
+
+    complete_count, incomplete_count, locked_count = 0, 0, 0
+    for section_key in block_data.get_children(course_usage_key):  # lint-amnesty, pylint: disable=too-many-nested-blocks
+        for subsection_key in block_data.get_children(section_key):
+            for unit_key in block_data.get_children(subsection_key):
+                complete = block_data.get_xblock_field(unit_key, 'complete', False)
+                contains_gated_content = block_data.get_xblock_field(unit_key, 'contains_gated_content', False)
```

Comment:
Just for my own sanity, will this always be False for verified learners? Want to make sure it isn't True, but they can access anyway and that affects the locked_count vs complete/incomplete counts.

Reply:
Good question I'll double check this

Reply:
So I traced how we use `contains_gated_content`, and it looks like this will be False for verified learners, but posting my logic here for a sanity check.

We use `contains_gated_content` in the `dates_banner_should_display()` logic, where we assign:
```
missed_gated_content = block_data.get_xblock_field(subsection_key, 'contains_gated_content', False)
```
and then use it to display the upgrade banner:
```
% if missed_gated_content:
            ${upgrade_to_reset_banner()}
```

So I think it's safe to use `contains_gated_content` to determine the locked count?

Reply:
👍 You should also be able to pretty easy validate locally by hitting the endpoint as an audit learner and then a verified learner and ensuring the verified learner doesn't see any of the gated content.